src/basic/terminal-util.c

```diff
@@ -455,7 +455,7 @@ int acquire_terminal(
                                         goto fail;
                                 }
 
-                                r = fd_wait_for_event(fd, POLLIN, ts + timeout - n);
+                                r = fd_wait_for_event(notify, POLLIN, ts + timeout - n);
```

Comment:
our fd is likely to be dead after the other process closed the tty if that other process was the controlling process of it. That's because ttys that lose their controlling process experience a "virtual hangup", which invalidates all other open fds to it. Background: if you have a modem line to a bash instance, and have backgrounded a few processes using bash's job control, and then you type "logout" into bash, then the tty link is hung up, and all of the backgrounded processes should also get a notified about that (either via SIGHUP — or if they blocked that by getting EIO when accessing the tty next).